strings/java_substring_comparisons/Solution.java

```diff
@@ -0,0 +1,27 @@
+import java.util.Scanner;
+import java.lang.String;
+
+public class Solution {
+  
+  public static String getSmallestAndLargest(String inputWord, int coefficient) {
+    String smallest = inputWord.substring(0, coefficient);
+    String largest = inputWord.substring(0, coefficient);
+    for (int i = 0; i <= inputWord.length() - coefficient; i++) {
+      if (inputWord.substring(i, i + coefficient).compareTo(smallest) < 0) {
```

Comment:
you're creating a substring 4 times on every iteration (though compiler might optimize it).
Instead you can save this substring and then use it.